internal/schema/graph.go

```diff
@@ -0,0 +1,120 @@
+package schema
+
+import mapset "github.com/deckarep/golang-set"
+
+import "encoding/json"
+
+// SchemaGraph represent the graph of a source
+type SchemaGraph struct {
+	Vertices mapset.Set
+	Edges    mapset.Set
+}
+
+// SchemaGraphJSON is the json representation of a schema graph
+type SchemaGraphJSON struct {
+	Vertices []AssetType    `json:"vertices"`
+	Edges    []RelationType `json:"edges"`
+}
+
+// NewSchemaGraph create a source graph
+func NewSchemaGraph() SchemaGraph {
+	return SchemaGraph{
+		Vertices: mapset.NewSet(),
+		Edges:    mapset.NewSet(),
+	}
+}
+
+// AddAsset add an asset type as a vertex in the source graph
+func (sg *SchemaGraph) AddAsset(assetType string) AssetType {
+	t := AssetType(assetType)
+	sg.Vertices.Add(t)
+	return t
+}
+
+// Assets return all the assets in the graph
+func (sg *SchemaGraph) Assets() []AssetType {
+	assets := make([]AssetType, 0)
```

Comment:
eq to `make([]AssetType)`, or just `[]AssetType{}`
if however you know the numbers or vertices, `make([]AssetType, 0, verticeCount)`

Reply:
solved